src/module.cpp

```diff
@@ -114,3 +114,12 @@ void Module::ApplyCompositions() {
 		compositions.pop_back();
 	}
 }
+
+void Module::VerifyFunction() {
+	for (const auto &input : inputSpecies) {
+		for (auto reaction : reactions) {
+			if (reaction.reactants[input] != reaction.products[input])
```

Comment:
I'm worried about an `std::out_of_range` exception here. What happens if you have a function
```
function MultiplyTwo {
    input: a;
    private: b;
    output: c;
    concentrations: {
        b := 2;
    }
    reactions: {
        a + b -> a + b + c;
        c -> 0;
    }
}
```
Or any other function, which contains a reaction that has no instance of all of the input species.

Reply:
Works fine here. I can implement a test to combat this in the feature?

Reply:
Huh, looking at the documentation for `std::map`, the [] operator does indeed insert a new member with the default value.

I'd say add the test anyway. Then if we switch away from maps or something, we won't have an unnoticed bug. Besides, more tests = more better;